### PR TITLE
allow use before define

### PR DESCRIPTION
**Pull request by turadg**
no ticket

## Description

The [no-use-before-define rule](https://eslint.org/docs/latest/rules/no-use-before-define) was motivated by when variables could hoist up in surprising ways. We always use `let` and `const` which have the Temporal Dead Zone that will throw an error if used before declaration.

So we don't need this rule anymore and it's mostly noise.

This also removes `no-constant-condition`. 

Whenever someone wants the rule they can still opt in,
```
/* eslint no-use-before-define: "error" */
```

### Security Considerations

n/a, lint

### Scaling Considerations

n/a, lint

### Documentation Considerations

reduces need for docs

### Testing Considerations

CI

### Upgrade Considerations

n/a, lint

### CODE REVIEW

**Comment by dckc** [comment]
> ... We always use `let` and `const`

We haven't removed all use of `function`. It looks like hundreds of them remain.

https://github.com/Agoric/agoric-sdk/blob/master/packages/SwingSet/src/controller/controller.js
https://github.com/Agoric/agoric-sdk/blob/master/packages/SwingSet/src/kernel/kernel.js
https://github.com/Agoric/agoric-sdk/blob/e4a60956afd8c8d4c830581b77d0daae4a493ae5/packages/vats/src/crc.js#L16
etc.

I don't think we even have a lint rule to avoid it.

**Comment by turadg** [comment]
> We haven't removed all use of `function`. It looks like hundreds of them remain.

Good point. But as you also point out this rule doesn't pertain to those.

Any reason to keep this rule?

**Comment by dckc** [comment]
> > We haven't removed all use of `function`. It looks like hundreds of them remain.
> 
> Good point. But as you also point out this rule doesn't pertain to those.

use-before-defined _does_ apply to functions, IIUC.

What I meant was: we don't have a rule to say "use `const` instead of `function`".

File: packages/orchestration/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+// Ambients
```
Comment (turadg) [inline packages/orchestration/src/types.ts]: oops, this in here is a mistake. I'll remove before merging

File: packages/governance/package.json [hunk 2]
```diff
@@ -34,7 +34,6 @@
     "@agoric/assert": "^0.6.0",
     "@agoric/ertp": "^0.16.2",
     "@agoric/internal": "^0.3.2",
-    "@agoric/network": "^0.1.0",
```
Comment (erights) [inline packages/governance/package.json]: An unrelated drive-by? (no objection. Just making sure I understand.)

File: packages/casting/package.json [hunk 3]
```diff
@@ -24,7 +24,6 @@
   "dependencies": {
     "@agoric/internal": "^0.3.2",
     "@agoric/notifier": "^0.6.2",
-    "@agoric/spawner": "^0.6.8",
```
Comment (erights) [inline packages/casting/package.json]: An unrelated drive-by? (no objection. Just making sure I understand.)

File: packages/governance/package.json (hunk 2 above)
Comment (turadg) [reply packages/governance/package.json]: yep, c4d7f8d250e1fe7693554705fda88cf0545ddf75, to improve the tred graph

**Comment by kriskowal** [comment]
I support this change even if we don’t fully migrate from `function` to `const` arrow functions. I happen to write a great deal of mutually recursive code that is best explained in the subjectively wrong order, so that is a bias, and I just don’t experience the surprise of using before definition _at runtime_ in practice. I can’t speak for everyone, though.

**Comment by dckc** [comment]
Does this mean I can go back to writing code top-down? oh my. whee!

**Comment by Chris-Hibbert** [comment]
I don't have a problem with these changes to our coding style.

> We always use let and const which have the Temporal Dead Zone that will throw an error if used before declaration.

Is use of `function` rather than `let` susceptible to the TDZ problem? That seems like an issue.

**Comment by erights** [comment]
As suggested, https://github.com/Agoric/agoric-sdk/pull/10464 would now globally suppress no-constant-condition, by borrowing that portion of this PR.

**Review by erights — APPROVED** [review]
As agreed verbally. LGTM, thanks!
